cmd/cdi-importer/importer.go

```diff
@@ -69,9 +71,15 @@ func main() {
 	}
 
 	dest := common.ImporterWritePath
+	if volumeMode == string(v1.PersistentVolumeBlock) &&
+		contentType == string(cdiv1.DataVolumeKubeVirt) && source == controller.SourceHTTP {
```

Comment:
Why only HTTP. I understand that this patch doesn't address registry yet, but now if I created a block pv with a registry annotation, its going try to write to the data space (which doesn't exist). That will will silently fail, and when we do get registry support and forget to update this (which is likely) that will fail as well.

So please either support registry in this patch, or do a check here and create an error appropriate for the situation so there is no way it can slip through the cracks.

Reply:
done

Reply:
I don't see this fixed.

Reply:
Two lines above:
if volumeMode == string(v1.PersistentVolumeBlock) && source != controller.SourceHTTP {
		klog.Errorf("Unsupported source type %s for block device", source)
		os.Exit(1)
	}

Reply:
Okay, since we are only supporting direct streaming from http into the block device, this is fine.